src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,172 @@
+#ifndef PHARE_SPLIT_2D_H
+#define PHARE_SPLIT_2D_H
+namespace PHARE::amr
+{
+template<> // Generic 2D delta setter
+struct SplitInnerSetter</*dim=*/2, 0>
+{
+    template<typename Deltas, typename DeltaVal>
+    static void set_deltas(Deltas& deltas, DeltaVal& delta_vals, size_t root)
+    {
+        for (size_t y = 0; y < root; y++)
+            for (size_t yoff = y * root, x = 0; x < root; x++)
+                deltas[x + yoff] = {delta_vals[x], delta_vals[y]};
+    }
+};
+
+
+/*************************************************************************
+  dim = 2
+  interp = 1
+  nbrRefineParticles = 9
+*/
+template<>
+class Splitter<2, 1, 9> : public ASplitter<2, 1, 9>
+{
+public:
+    Splitter()
+        : ASplitter{weight_vals_, delta_vals_}
+    {
+    }
+
+protected:
+    constexpr static float delta_vals_[]  = {-1, 0, 1};
+    constexpr static float weight_vals_[] = {0.25, 0.125, 0.01625};
+};
+
+template<>
+struct SplitInnerSetter</*dim=*/2, /*nbrRefineParticles*/ 9>
+{
+    constexpr static size_t dim  = 2;
+    constexpr static size_t root = 3; // 3*3
+
+    template<typename Weights, typename WeightsVal>
+    static void set_weights(Weights& weights, WeightsVal& weight_vals)
```

Comment:
this function is a bit hard to read, at this point I've not read the 3D equivalent yet but I'm scared :)
how can we ease the reading so that it becomes easy to get what  0,2,6,8 are and why they get weight val 2, same for 1,3,5,7 and weight val 1, and why particle 4 gets weight 0

Reply:
I don't have an answer for this,

Reply:
perhaps something like thishttps://youtu.be/IAdLwUXRUvg?t=744

![image](https://user-images.githubusercontent.com/30491/81413462-c14c5e80-9145-11ea-98c7-6314e6368c11.png)

Reply:
this is not in the standard, does not compile with our flags, and is actually quite particular insofar as I believe each weights per exact split would require it's own enum.

Reply:
I have provided an alternative approach here: https://github.com/Dekken/PHARE/blob/r2d2/src/amr/data/particles/refine/split_2d.h#L33